README.md

```diff
@@ -195,6 +195,106 @@ It will be slow in the first couple of calls as it loads the model in a lazy man
 Voila! That's a proof that this Deep Learning model is going to be trained and deployed on AWS SageMaker successfully. Now, go to the *Usage* section in [Sagify Docs](https://Kenza-AI.github.io/sagify/) to see how to train and deploy this Deep Learning model to AWS SageMaker!
 
 
+## Hyperparameter Optimization
+
+Given that you have configured your AWS Account as described in the previous section, you're now ready to perform Bayesian Hyperparameter Optimization on AWS SageMaker! The process is similar to training step.
+
+### Step 1: Define Hyperparameter Configuration File
+
+Define the Hyperparameter Configuration File. More specifically, you need to specify in a local JSON file the ranges for the hyperparameters, the name of the objective metric and its type (i.e. `Maximize` or `Minimize`). For example:
+
+```
+{
+	"ParameterRanges": {
+		"CategoricalParameterRanges": [
+			{
+				"Name": "kernel",
+				"Values": ["linear", "rbf"]
+			}
+		],
+		"ContinuousParameterRanges": [
+		{
+		  "MinValue": 0.001,
+		  "MaxValue": 10,
+		  "Name": "gamma"
+		}
+		],
+		"IntegerParameterRanges": [
+			{
+				"Name": "C",
+				"MinValue": 1,
+				"MaxValue": 10
+			}
+		]
+    },
+    "ObjectiveMetric": {
+    	"Name": "Precision",
+        "Type": "Maximize"
+    }
+}
+```
+
+### Step 2: Implement Train function
+
+Replace the `TODOs` in the `try..except` of `train(...)` function in `sagify/training/train` file with your logic. For example:
```

Comment:
Just a note, and I think we have discussed this before, I think we should next work on extracting that code into its own function that we then call / inject from `sagify` instead of asking of users to edit existing files.

Reply:
Yeah, I'll add it to issues ;-)